tests/unit/test_oidc.py

```diff
@@ -0,0 +1,283 @@
+"""
+Unit tests for the oidc code in the frontend.
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+import socket
+import mock
+import urlparse
+
+import oic
+import oic.oic.message as message
+
+import ga4gh.frontend as frontend
+import tests.utils as utils
+
+RANDSTR = 'abc'
+OICCODE = (
+    "5GzAsW4Gfz8Km8PTuayFi6j2tA2VnPrHiWybL%2BkryEpr05D0OUKTGTi49Z54yl%2"
+    "FIB5KsUzGwSg%2BFpx%2FpBt0aVc3QsvPlsJu6WGHIchJ55UU%3D")
+
+OICRESPONSE = {
+    'scope': 'openid profile',
+    'state': RANDSTR,
+    'code': '5GzAsW4Gfz8Km8PTuayFi6j2tA2VnPrHiWybL+'
+            'kryEpr05D0OUKTGTi49Z54yl/IB5KsUzGwSg+'
+            'Fpx/pBt0aVc3QsvPlsJu6WGHIchJ55UU='}
+
+
+def mockAuthorizationRequest(self, request_args, state):
+    class mockResult:
+        def __init__(self):
+            self.url = None
+
+    r = mockResult()
```

Comment:
same comment as last file... avoid single-letter and double-letter variable names... they're less descriptive